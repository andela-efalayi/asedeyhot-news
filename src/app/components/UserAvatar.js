import React from 'react';
import Avatar from 'material-ui/Avatar';
import ListItem from 'material-ui/List/ListItem';


const avatarStyle = {
  right: 1
};

const UserAvatar = (props) => (
   <ListItem
    primaryText={props.userName}
    insetChildren={true}
    rightAvatar={<Avatar src={props.userImg}
    style={avatarStyle} />}
  />
);

export default UserAvatar;
