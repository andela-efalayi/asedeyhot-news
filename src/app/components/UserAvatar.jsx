/* global localStorage window */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Avatar from 'material-ui/Avatar';
import ListItem from 'material-ui/List/ListItem';
import IconButton from 'material-ui/IconButton';

import Popover, { PopoverAnimationVertical } from 'material-ui/Popover';
import Menu from 'material-ui/Menu';
import MenuItem from 'material-ui/MenuItem';
import PowerSettingsNew from 'material-ui/svg-icons/action/power-settings-new';

const avatarStyle = {
  marginTop: '-15px'
};

const styles = {
  signoutPopOverMenu: {
    marginTop: '-15px',
    marginBottom: '0px',
    right: '0px',
    fontSize: '13px'
  }
};

class UserAvatar extends Component {
  constructor(props) {
    super(props);

    this.state = {
      open: false
    };
  }

  handleTouchTap = (event) => {
    event.preventDefault();
    this.setState({
      open: true,
      anchorEl: event.currentTarget,
    });
  };

  handleRequestClose = () => {
    this.setState({
      open: false,
    });
  };

  signOutUser = () => {
    localStorage.clear();
    window.location = '/';
  }
  render() {
    return (
      <ListItem
      primaryText={this.props.user.name}
      insetChildren
      hoverColor="none"
      rightAvatar={
        <IconButton
          tooltip="click to sign out"
          tooltipPosition="top-center"
          disableTouchRipple
          iconStyle={avatarStyle}
          onTouchTap={this.handleTouchTap}
          onClick={this.handleTouchTap}>
          <Avatar src={this.props.user.imageUrl}/>
        </IconButton>
      }>
      <Popover
          className="signOutPopOver"
          open={this.state.open}
          anchorEl={this.state.anchorEl}
          anchorOrigin={{ horizontal: 'left', vertical: 'bottom' }}
          targetOrigin={{ horizontal: 'left', vertical: 'top' }}
          animation={PopoverAnimationVertical}
          onRequestClose={this.handleRequestClose}
        >
          <Menu maxHeight={28} desktop={false}
          onTouchTap={this.signOutUser}
          onClick={this.signOutUser}>
            <MenuItem primaryText="Sign out"
            style={styles.signoutPopOverMenu}
            rightIcon={<PowerSettingsNew />}
            value="signout" />
          </Menu>
        </Popover>
    </ListItem>
    );
  }
}

UserAvatar.propTypes = {
  user: PropTypes.object
};
UserAvatar.defaultProps = {
  user: null
};

export default UserAvatar;
