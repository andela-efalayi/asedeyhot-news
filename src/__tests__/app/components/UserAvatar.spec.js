import React from 'react';
import { shallow } from 'enzyme';
import UserAvatar from '../../../app/components/UserAvatar.jsx';

const user = {
  name: 'Jest Tester',
  imageUrl: 'https://dummyimage.com/600x400/000000/0011ff'
};

describe('UserAvatar Component', () => {
  it('should render an avatar for a user', () => {
    const wrapper = shallow(
        <UserAvatar user={user}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
