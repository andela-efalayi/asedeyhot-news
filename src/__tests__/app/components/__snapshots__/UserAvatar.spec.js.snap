// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar Component should render an avatar for a user 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ListItem
    autoGenerateNestedIndicator={true}
    containerElement="span"
    disableKeyboardFocus={false}
    disabled={false}
    hoverColor="none"
    initiallyOpen={false}
    insetChildren={true}
    nestedItems={Array []}
    nestedLevel={0}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onNestedListToggle={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    open={null}
    primaryText="Jest Tester"
    primaryTogglesNestedList={false}
    rightAvatar={
        <IconButton
            disableTouchRipple={true}
            disabled={false}
            iconStyle={
                Object {
                    "marginTop": "-15px",
                  }
            }
            onTouchTap={[Function]}
            tooltip="click to sign out"
            tooltipPosition="top-center"
            touch={false}
        >
            <Avatar
                size={40}
                src="https://dummyimage.com/600x400/000000/0011ff"
            />
        </IconButton>
    }
    secondaryTextLines={1}
>
    <Popover
        anchorEl={undefined}
        anchorOrigin={
            Object {
                "horizontal": "left",
                "vertical": "bottom",
              }
        }
        animated={true}
        animation={[Function]}
        autoCloseWhenOffScreen={true}
        canAutoPosition={true}
        className="signOutPopOver"
        onRequestClose={[Function]}
        open={false}
        style={
            Object {
                "overflowY": "auto",
              }
        }
        targetOrigin={
            Object {
                "horizontal": "left",
                "vertical": "top",
              }
        }
        useLayerForClickAway={true}
        zDepth={1}
    >
        <Menu
            autoWidth={true}
            desktop={false}
            disableAutoFocus={false}
            initiallyKeyboardFocused={false}
            maxHeight={28}
            multiple={false}
            onChange={[Function]}
            onEscKeyDown={[Function]}
            onItemTouchTap={[Function]}
            onKeyDown={[Function]}
        >
            <MenuItem
                anchorOrigin={
                    Object {
                        "horizontal": "right",
                        "vertical": "top",
                      }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                primaryText="Sign out"
                rightIcon={<ActionPowerSettingsNew />}
                style={
                    Object {
                        "fontSize": "13px",
                        "marginBottom": "0px",
                        "marginTop": "-15px",
                        "right": "0px",
                      }
                }
                value="signout"
            />
        </Menu>
    </Popover>
</ListItem>,
  "nodes": Array [
    <ListItem
      autoGenerateNestedIndicator={true}
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      hoverColor="none"
      initiallyOpen={false}
      insetChildren={true}
      nestedItems={Array []}
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Jest Tester"
      primaryTogglesNestedList={false}
      rightAvatar={
            <IconButton
                  disableTouchRipple={true}
                  disabled={false}
                  iconStyle={
                        Object {
                              "marginTop": "-15px",
                            }
                  }
                  onTouchTap={[Function]}
                  tooltip="click to sign out"
                  tooltipPosition="top-center"
                  touch={false}
            >
                  <Avatar
                        size={40}
                        src="https://dummyimage.com/600x400/000000/0011ff"
                  />
            </IconButton>
      }
      secondaryTextLines={1}
>
      <Popover
            anchorEl={undefined}
            anchorOrigin={
                  Object {
                        "horizontal": "left",
                        "vertical": "bottom",
                      }
            }
            animated={true}
            animation={[Function]}
            autoCloseWhenOffScreen={true}
            canAutoPosition={true}
            className="signOutPopOver"
            onRequestClose={[Function]}
            open={false}
            style={
                  Object {
                        "overflowY": "auto",
                      }
            }
            targetOrigin={
                  Object {
                        "horizontal": "left",
                        "vertical": "top",
                      }
            }
            useLayerForClickAway={true}
            zDepth={1}
      >
            <Menu
                  autoWidth={true}
                  desktop={false}
                  disableAutoFocus={false}
                  initiallyKeyboardFocused={false}
                  maxHeight={28}
                  multiple={false}
                  onChange={[Function]}
                  onEscKeyDown={[Function]}
                  onItemTouchTap={[Function]}
                  onKeyDown={[Function]}
            >
                  <MenuItem
                        anchorOrigin={
                              Object {
                                    "horizontal": "right",
                                    "vertical": "top",
                                  }
                        }
                        checked={false}
                        desktop={false}
                        disabled={false}
                        focusState="none"
                        insetChildren={false}
                        primaryText="Sign out"
                        rightIcon={<ActionPowerSettingsNew />}
                        style={
                              Object {
                                    "fontSize": "13px",
                                    "marginBottom": "0px",
                                    "marginTop": "-15px",
                                    "right": "0px",
                                  }
                        }
                        value="signout"
                  />
            </Menu>
      </Popover>
</ListItem>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <UserAvatar
        user={
                Object {
                        "imageUrl": "https://dummyimage.com/600x400/000000/0011ff",
                        "name": "Jest Tester",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": UserAvatar {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleRequestClose": [Function],
        "handleTouchTap": [Function],
        "props": Object {
          "user": Object {
            "imageUrl": "https://dummyimage.com/600x400/000000/0011ff",
            "name": "Jest Tester",
          },
        },
        "refs": Object {},
        "signOutUser": [Function],
        "state": Object {
          "open": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ListItem
          autoGenerateNestedIndicator={true}
          containerElement="span"
          disableKeyboardFocus={false}
          disabled={false}
          hoverColor="none"
          initiallyOpen={false}
          insetChildren={true}
          nestedItems={Array []}
          nestedLevel={0}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onNestedListToggle={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          open={null}
          primaryText="Jest Tester"
          primaryTogglesNestedList={false}
          rightAvatar={
                    <IconButton
                              disableTouchRipple={true}
                              disabled={false}
                              iconStyle={
                                        Object {
                                                  "marginTop": "-15px",
                                                }
                              }
                              onTouchTap={[Function]}
                              tooltip="click to sign out"
                              tooltipPosition="top-center"
                              touch={false}
                    >
                              <Avatar
                                        size={40}
                                        src="https://dummyimage.com/600x400/000000/0011ff"
                              />
                    </IconButton>
          }
          secondaryTextLines={1}
>
          <Popover
                    anchorEl={undefined}
                    anchorOrigin={
                              Object {
                                        "horizontal": "left",
                                        "vertical": "bottom",
                                      }
                    }
                    animated={true}
                    animation={[Function]}
                    autoCloseWhenOffScreen={true}
                    canAutoPosition={true}
                    className="signOutPopOver"
                    onRequestClose={[Function]}
                    open={false}
                    style={
                              Object {
                                        "overflowY": "auto",
                                      }
                    }
                    targetOrigin={
                              Object {
                                        "horizontal": "left",
                                        "vertical": "top",
                                      }
                    }
                    useLayerForClickAway={true}
                    zDepth={1}
          >
                    <Menu
                              autoWidth={true}
                              desktop={false}
                              disableAutoFocus={false}
                              initiallyKeyboardFocused={false}
                              maxHeight={28}
                              multiple={false}
                              onChange={[Function]}
                              onEscKeyDown={[Function]}
                              onItemTouchTap={[Function]}
                              onKeyDown={[Function]}
                    >
                              <MenuItem
                                        anchorOrigin={
                                                  Object {
                                                            "horizontal": "right",
                                                            "vertical": "top",
                                                          }
                                        }
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        primaryText="Sign out"
                                        rightIcon={<ActionPowerSettingsNew />}
                                        style={
                                                  Object {
                                                            "fontSize": "13px",
                                                            "marginBottom": "0px",
                                                            "marginTop": "-15px",
                                                            "right": "0px",
                                                          }
                                        }
                                        value="signout"
                              />
                    </Menu>
          </Popover>
</ListItem>,
        "_debugID": 2,
        "_renderedOutput": <ListItem
          autoGenerateNestedIndicator={true}
          containerElement="span"
          disableKeyboardFocus={false}
          disabled={false}
          hoverColor="none"
          initiallyOpen={false}
          insetChildren={true}
          nestedItems={Array []}
          nestedLevel={0}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onNestedListToggle={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          open={null}
          primaryText="Jest Tester"
          primaryTogglesNestedList={false}
          rightAvatar={
                    <IconButton
                              disableTouchRipple={true}
                              disabled={false}
                              iconStyle={
                                        Object {
                                                  "marginTop": "-15px",
                                                }
                              }
                              onTouchTap={[Function]}
                              tooltip="click to sign out"
                              tooltipPosition="top-center"
                              touch={false}
                    >
                              <Avatar
                                        size={40}
                                        src="https://dummyimage.com/600x400/000000/0011ff"
                              />
                    </IconButton>
          }
          secondaryTextLines={1}
>
          <Popover
                    anchorEl={undefined}
                    anchorOrigin={
                              Object {
                                        "horizontal": "left",
                                        "vertical": "bottom",
                                      }
                    }
                    animated={true}
                    animation={[Function]}
                    autoCloseWhenOffScreen={true}
                    canAutoPosition={true}
                    className="signOutPopOver"
                    onRequestClose={[Function]}
                    open={false}
                    style={
                              Object {
                                        "overflowY": "auto",
                                      }
                    }
                    targetOrigin={
                              Object {
                                        "horizontal": "left",
                                        "vertical": "top",
                                      }
                    }
                    useLayerForClickAway={true}
                    zDepth={1}
          >
                    <Menu
                              autoWidth={true}
                              desktop={false}
                              disableAutoFocus={false}
                              initiallyKeyboardFocused={false}
                              maxHeight={28}
                              multiple={false}
                              onChange={[Function]}
                              onEscKeyDown={[Function]}
                              onItemTouchTap={[Function]}
                              onKeyDown={[Function]}
                    >
                              <MenuItem
                                        anchorOrigin={
                                                  Object {
                                                            "horizontal": "right",
                                                            "vertical": "top",
                                                          }
                                        }
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        primaryText="Sign out"
                                        rightIcon={<ActionPowerSettingsNew />}
                                        style={
                                                  Object {
                                                            "fontSize": "13px",
                                                            "marginBottom": "0px",
                                                            "marginTop": "-15px",
                                                            "right": "0px",
                                                          }
                                        }
                                        value="signout"
                              />
                    </Menu>
          </Popover>
</ListItem>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UserAvatar
    user={
        Object {
            "imageUrl": "https://dummyimage.com/600x400/000000/0011ff",
            "name": "Jest Tester",
          }
    }
/>,
}
`;
